lib/asyn-thread.c

```diff
@@ -135,7 +135,7 @@ void Curl_resolver_cleanup(void *resolver)
 int Curl_resolver_duphandle(void **to, void *from)
 {
   (void)from;
-  return Curl_resolver_init(to);
+  return Curl_resolver_init(NULL, to);
```

Comment:
This looks wrong. Shouldn't this then pass in the new easy handle here? Otherwise this stores a NULL pointer to get used in the callback later on, and the callback doesn't deal with a NULL pointer...

Reply:
Ok, there is a problem for c-ares to.
In case `Curl_resolver_duphandle` is called, `ares_dup` is then called, but the `sock_state_cb_data`  of the new ares handle should be changed to point to the newly allocated easy. 
Right now c-ares only supports setting the `sock_state_cb_data` in `init_by_options` so the only way that I see how to fix it now, is to create a new ares handle instead of using `ares_dup`
What do you think ?